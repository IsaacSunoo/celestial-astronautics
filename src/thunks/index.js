import { setLoading, getApolloData, setError, getDaily } from '../actions';
import { fetchCall } from '../api';
import apiKey from '../api_key/apiKey';

// const baseUrl = 'https://api.nasa.gov/';

export const getApollo20 = () => {
    const url = 'https://images-api.nasa.gov/search?q=apollo%20landing&media_type=image';
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const apollo = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getApolloData(apollo));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

export const getDailyNews = () => {
    const url = `https://api.nasa.gov/planetary/apod?api_key=${apiKey}`;
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const dailyNews = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getDaily(dailyNews));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

