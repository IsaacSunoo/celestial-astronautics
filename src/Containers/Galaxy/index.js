import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getGalaxy } from '../../thunks';
import LoadingGif from '../../images/Loading.gif';
import InfoCard from '../../components/InfoCard';
import Particles from '../../components/Particles';
import PropTypes from 'prop-types';

export class Galaxy extends Component {
    componentDidMount = async () => {
        const { getGalaxy } = this.props;
        await getGalaxy();
    }

    render() {
        const { galaxy, isLoading } = this.props;
        let displayGalaxy;
        if (galaxy.length !== 0) {
            displayGalaxy = galaxy.items.map(item => {
            return (
                <div key={item.data[0].nasa_id}>
                    <InfoCard information={item} />
                </div>
                )
            });
        }

        return (
            isLoading ?
            (
            <div className='loading-display'>
                <h2>Loading</h2>
                <img src={LoadingGif} alt='loading gif' />
            </div>
            ) :
            (
            <div className='info-items-container'>
                <Particles />
                {displayGalaxy}
            </div>
            )
        )
    }
}

export const mapStateToProps = state => ({
    galaxy: state.galaxy,
    isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
    getGalaxy: () => dispatch(getGalaxy())
});

Galaxy.propTypes = {
  getGalaxy: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
};


export default connect(mapStateToProps, mapDispatchToProps)(Galaxy);