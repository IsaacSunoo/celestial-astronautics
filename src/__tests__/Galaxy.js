import React from 'react';
import { Galaxy, mapStateToProps, mapDispatchToProps } from '../containers/Galaxy';
import { shallow } from 'enzyme';
import { getGalaxy } from '../thunks';

jest.mock('../thunks');
const mockGalaxy = [];
const mockIsLoading = false;

describe('galaxy', () => {
  let wrapper;
  window.fetch = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Galaxy galaxy={mockGalaxy} isLoading={mockIsLoading} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it.skip('returns an object with galaxies', () => {
      const mockState = {
        galaxy: ['jupiter', 'saturn'],
        isLoading: false,
        error: 'error'
      };
      const expectedState = {
        galaxy: ['jupiter', 'saturn'],
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a getGalaxy thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getGalaxy();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getGalaxy();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
