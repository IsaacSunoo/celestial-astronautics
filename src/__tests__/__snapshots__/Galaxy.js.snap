// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`galaxy should match snapshot 1`] = `
<div
  className="info-items-container"
>
  <ParticlesContainer />
</div>
`;
